### ISSUE TITLE
Solving a system of coupled PDEs

### ISSUE THREAD

**Issue by AJAXJR24** [lululxvi/deepxde#1503]
Dear @lululxvi 
Thanks for your time
I'm solving a system consisting of a PDE and ODE which are coupled together.
I got underlying problems:
1) for the second equation the Temperature (y[0:1]) must be the average temperature in the whole domain. Is defining the way I did it ok?
2) I can't minimize the loss of the Robin boudary condithion and the first initial condition. I don't know what to do. 

`Le` = 0.013 # m
He = 0.065 # m
kl = 140 # Thermal conductivity in angular direction (W/mK)
kh = 0.7 # Thermal conductivity in radial direction (W/mk)
T0 = 273.15 + 20 # Initial Temperature (C)
rho = 2231.2 # Cell density (kg/m^3)
Cp = 1100 # Heat capacity (J/kgK)
h = 20 # Heat transfer coefficient (W/m^2K)
Rc = 8.314 # Gas constant (J/molK)
E = 2.7e5 # Activation energy value of electrolyte [J/mol]
A = 5.14e25 # Reaction factor of electrolyte [1/s]
m = 1 # Reaction order of electrolyte
c0 = 1 # Initial dimensionless concentration
t_max = 60 * 60 # solution time
H = 6.2e5
W = 335

geom = dde.geometry.geometry_2d.Rectangle([0,0],[Le,He])
timedomain = dde.geometry.TimeDomain(0,t_max)
geomtime = dde.geometry.GeometryXTime(geom, timedomain)

def pde(x,y):
  #T,c = y[:,0:1],y[:,1:2]
  T_xx = dde.grad.hessian(y,x,component=0,i=0,j=0)
  T_yy = dde.grad.hessian(y,x,component=0,i=1,j=1)
  T_t = dde.grad.jacobian(y,x,i=0,j=2)
  c_t = dde.grad.jacobian(y,x,i=1,j=2)
  ###########################################
  R = -c_t
  ###########################################
  print(y[:,0:1])
  return [(rho*Cp*T_t) - (kl*T_xx) - (kh*T_yy) - (H * W * R),
          A * tf.exp(-E/(Rc*tf.reduce_mean(y[:,0:1]))) * y[:,1:2] + c_t]

def boundary_r(x, on_boundary):
    return on_boundary and dde.utils.isclose(x[0], Le)
def boundary_l(x, on_boundary):
    return on_boundary and dde.utils.isclose(x[0], 0)
def boundary_d(x, on_boundary):
    return on_boundary and dde.utils.isclose(x[1], 0)
def boundary_u(x, on_boundary):
    return on_boundary and dde.utils.isclose(x[1], He)

def func_conv(x,y):
  T_ext = 250 + 273.15
  return (h * (T_ext-y[:,0:1])/-kl).reshape(-1,1)`

bc_d = dde.icbc.NeumannBC(geomtime, lambda x: 0,boundary_d, component = 0)
bc_u = dde.icbc.NeumannBC(geomtime, lambda x: 0,boundary_u, component = 0)
bc_l = dde.icbc.NeumannBC(geomtime, lambda x: 0,boundary_l, component = 0)
bc_r = dde.icbc.RobinBC(geomtime, func_conv , boundary_r, component = 0)

`ic1 = dde.icbc.IC(geomtime,lambda X: T0 ,lambda _, on_initial: on_initial, component = 0)
ic2 = dde.icbc.IC(timedomain,lambda X: 1 ,lambda _, on_initial: on_initial, component = 1)

data = dde.data.TimePDE(
    geomtime,
    pde,
    [bc_l,bc_d,bc_u,bc_r, ic1,ic2],
    num_domain=100000,
    num_boundary=80000,
    num_initial=80000,
    )

net = dde.nn.FNN([3] + [25] * 6 + [2], "tanh", "Glorot normal")
model = dde.Model(data, net)
model.compile("adam", lr=1e-3)
losshistory, train_state = `model.train(iterations=2e+05)`
 Thanks
